dotCMS/src/main/java/com/dotcms/visitor/filter/logger/VisitorLogger.java

```diff
@@ -118,24 +119,42 @@ public static List<Constructor<AbstractCharacter>> removeConstructor(
     }
 
     private static void logInternal(final HttpServletRequest request, final HttpServletResponse response)
-            throws NoSuchMethodException, SecurityException, InstantiationException, IllegalAccessException,
+            throws SecurityException, InstantiationException, IllegalAccessException,
             IllegalArgumentException, InvocationTargetException {
 
-        AbstractCharacter base = new BaseCharacter(request, response);
         try {
-            for (Constructor<AbstractCharacter> con : constructors) {
-                base = con.newInstance(base);
-            }
-
-            for (Constructor<AbstractCharacter> con : customConstructors) {
-                base = con.newInstance(base);
-            }
+            final Character base = createCharacter(request, response);
```

Comment:
why we need this changes here?

Reply:
i have encapsulate this code into a method, in order to be reused into another part